src/core/server/http/router/request.ts

```diff
@@ -97,17 +104,26 @@ export class KibanaRequest<Params = unknown, Query = unknown, Body = unknown> {
     request: Request,
     readonly params: Params,
     readonly query: Query,
-    readonly body: Body
+    readonly body: Body,
+    private readonly secured: boolean
   ) {
-    this.headers = request.headers;
     this.url = request.url;
 
-    this[requestSymbol] = request;
+    // prevent Symbol exposure via Object.getOwnPropertySymbols()
+    Object.defineProperty(this, requestSymbol, {
```

Comment:
Probably, we should make KibanaRequest an object literal instead of class to emulate private fields via closures. Also, the class exposes static methods are meant to be used only inside of the core.

Reply:
That sounds like a smart move to me, can be done in future PR.